### PR TITLE
Move doc files into resources and fix README links

### PR DESCRIPTION
**Pull request by DeLaGuardo**
> Related to https://github.com/advthreat/iroh/issues/7273

Move documentation-related files into the resource folder so they wouldn't conflict with dependencies.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
public: changed directory structure `/doc` moved to `/resources/ctia/doc`
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/documentation/routes.clj [hunk 1]
```diff
@@ -75,7 +77,7 @@
 (defn render-request
   "read the requested file from resources, render it if needed"
   [path-info]
-  (let [file-path (subs path-info 1)
+  (let [file-path (str doc-resource-prefix path-info)
         file-content (get-file-content file-path)
```
Comment (msprunck) [inline src/ctia/documentation/routes.clj:81]: Could you tighten the security of the function `get-file-content` in the same way [`ring.util.response/resource-response`](https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/util/response.clj#L324) is? It avoids:
- directory traversal
- it doesn't follow symlinks if specified

File: src/ctia/documentation/routes.clj [hunk 2]
```diff
@@ -49,49 +45,53 @@
...
+                 (get :body)
+                 (slurp)
+                 (md-to-html-string)
+                 (decorate-markdown))
+        body-length (count (.getBytes body "UTF-8"))]
+    (-> resp
+        (assoc :body body)
+        (response/content-type "text/html")
+        (response/charset "UTF-8")
+        (response/update-header "Content-Length" (constantly (str body-length))))))
```
Comment (msprunck) [inline src/ctia/documentation/routes.clj:73]:
```suggestion
        (response/header "Content-Length" (str body-length)))))
```

Comment (DeLaGuardo) [reply src/ctia/documentation/routes.clj]: Thanks! No idea how I oversee `header` :) It's even stringifying the content automatically

**Comment by NeelSaini** [comment]
Cherry Picked on TEST in rel v1.109

**Pull request merged**